src/v/resource_mgmt/tests/cpu_profiler_test.cc

```diff
@@ -40,6 +40,28 @@ ss::future<> busy_loop(std::chrono::milliseconds duration) {
         co_await ss::coroutine::maybe_yield();
     }
 }
+
+// Creates a sharded_profiler, and destroys it when this object
+// is destructed (must be run in ss::thread).
+struct cp_holder {
```

Comment:
There are a few ways to go to ensure service desctruction: using RAII and THREAD_TEST_CASE is the way I went.